File: parse_errors.js
const fs = require('fs');
const path = require('path');

try {
  const data = JSON.parse(fs.readFileSync('eslint-report.json', 'utf8'));
  for (const fileInfo of data) {
    const filePath = fileInfo.filePath;
    const relPath = path.relative(process.cwd(), filePath);
    const messages = fileInfo.messages || [];
    if (messages.length > 0) {
      console.log(`FILE: ${relPath}`);
      for (const msg of messages) {
        console.log(`  Line ${msg.line}:${msg.column} - [${msg.ruleId}] - ${msg.message}`);
      }
      console.log('');
    }
  }
} catch (err) {
  console.error(err);
}

File: parse_errors.py
import json
import os

with open("eslint-report.json", "r", encoding="utf-8") as f:
    data = json.load(f)

for file_info in data:
    filepath = file_info.get("filePath", "")
    # Make filepath relative to the repo root if possible
    rel_path = os.path.relpath(filepath, os.getcwd()) if filepath else ""
    messages = file_info.get("messages", [])
    if messages:
        print(f"FILE: {rel_path}")
        for msg in messages:
            line = msg.get("line")
            col = msg.get("column")
            rule = msg.get("ruleId")
            text = msg.get("message")
            print(f"  Line {line}:{col} - [{rule}] - {text}")
        print()
